cmds/core/sluinit/uinit.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"github.com/u-root/u-root/pkg/cmdline"
+	slaunch "github.com/u-root/u-root/pkg/securelaunch"
+	"github.com/u-root/u-root/pkg/securelaunch/policy"
+	"github.com/u-root/u-root/pkg/securelaunch/tpm"
+	"log"
+	"os"
+)
+
+func main() {
+
+	slaunch.Debug("********Step 1: init completed. starting main ********")
+	tpmDev, err := tpm.GetHandle()
+	if err != nil {
+		log.Printf("tpm.getHandle failed. err=%v", err)
+		os.Exit(1)
+	}
+	defer tpmDev.Close()
+
+	slaunch.Debug("********Step 2: locate SL Policy ********")
+	rawBytes, err := policy.Locate()
+	if err != nil {
+		log.Printf("locate SL Policy failed: err=%v\n", err)
+		os.Exit(1)
+	}
+	slaunch.Debug("policy file located\n")
+
+	slaunch.Debug("********Step 3: parse SL Policy ********")
+	//TODO: The policy file must be measured and extended into PCR21 (PCR15
+	// until DRTM launch is working and able to set locality
+	p, err := policy.Parse(rawBytes)
+	if err != nil {
+		log.Printf("parse Policy failed")
+		os.Exit(1)
+	}
+	if p == nil {
+		log.Printf("SL Policy parsed into a null set")
+		os.Exit(1)
+	}
+	slaunch.Debug("policy file parsed\n")
+
+	slaunch.Debug("********Step 4: Collecting Evidence ********")
+	slaunch.Debug("policy file parsed=%v\n", p)
+
+	for _, c := range p.Collectors {
+		slaunch.Debug("Input Collector: %v\n", c)
+		c.Collect(tpmDev)
+	}
+	slaunch.Debug("Collectors completed\n")
+
+	slaunch.Debug("********Step 5: Write eventlog to /boot partition*********")
+	if e := p.EventLog.Persist(); e != nil {
+		log.Printf("EventLog.Persist() failed err=%v", e)
+		os.Exit(1)
+	}
+
+	slaunch.Debug("********Step 6: Launcher called ********")
+	err = p.Launcher.Boot(tpmDev)
+	log.Printf("Boot failed. err=%s", err)
+}
+
+func init() {
+	// check if uroot-debug=true in kernel cmdline.
+	val, ok := cmdline.Flag("uroot-debug")
```

Comment:
I'd actually prefer if this was a regular flag like anything else. Then you can run `sluinit -v` for debugging locally, too.

See https://github.com/u-root/u-root/blob/d70990c99c10d0eb6becb8ddd29b074cdde1ae1b/cmds/core/init/init.go#L76

You can specify it via the cmdline as uroot.uinitargs="-v -other-arg"

cmds/core/init will automatically pass those as args to uinit.

Does that solve your use case or is it too weird?

Reply:
Hmm, we thought about a uroot flag but mostly issues in our lab is that we want to live debug the system with securelaunch kernel installed. For example the user only has iscsi drives and he forgot to pass netroot cmdline parameter in kernel, that would mean sluinit will fail and won’t let it boot but if he passes uroot-debug=true,  he can just reboot the kernel by adding this parameter in grub, realize that the issue is because of iscsi and can still boot without reinstalling the kernel. 

Also it saves us lot of time during testing live bugs etc.

Reply:
I'm not sure I understand your objection -- you can still use the kernel cmdline, as long as you put 

uroot.uinitargs=-v

in there

Reply:
Ah, got u. I thought you meant the u-root command line.